src/circuit/gadget/sinsemilla/chip/hash_to_point.rs

```diff
@@ -24,7 +21,7 @@ impl SinsemillaChip {
     pub(super) fn hash_message(
         &self,
         region: &mut Region<'_, pallas::Base>,
-        Q: pallas::Affine,
+        Q: (usize, pallas::Affine), // This tuple contains: (absolute offset of Q_x in public inputs column, Q)
```

Comment:
Hmm, this breaks the reusable chip abstraction: chips should not depend on anything being in particular absolute rows, as that is not portable between different circuits. I'm not sure it's something we can prevent in the `halo2` instance API (since the top-level circuit builder _does_ need to gain access to public inputs somewhere), but its usage should be discouraged for internal chips.

Instead, we should implement https://github.com/zcash/halo2/issues/335, which would make the positioning of constants a layouter concern.

Reply:
See also https://github.com/zcash/halo2/pull/341#pullrequestreview-710410868.

Reply:
This has been addressed by using https://github.com/zcash/halo2/pull/342.